lib/resourcemerge/machineconfig.go

```diff
@@ -78,6 +79,37 @@ func ensureControllerConfigSpec(modified *bool, existing *mcfgv1.ControllerConfi
 	setBytesIfSet(modified, &existing.KubeAPIServerServingCAData, required.KubeAPIServerServingCAData)
 	setBytesIfSet(modified, &existing.CloudProviderCAData, required.CloudProviderCAData)
 
+	// First off: do I need to generally account for Infra changes? Seems like we should?
+	if required.Infra != nil && !equality.Semantic.DeepEqual(existing.Infra, required.Infra) {
+		*modified = true
+		existing.Infra = required.Infra
+	}
+
+	// Moving on... if all Infras are empty (ie born in 4.1 clusters), maybe update Infra with values we _do_ have(?)
+	if required.Infra == nil && existing.Infra == nil {
```

Comment:
`required` can't be empty here as it's the one that we bake into the CC in the MCO sync - `existing` OTOH can indeed be nil if you're coming from 4.1

Reply:
in other words, the check at line 83 makes sure, at this point, that none of those 2 are empty anymore. If CC.infra (existing) was nil, line 85 populated it since required is never nil when this code runs

Reply:
Ahh yes I had these in a diff order and didn't reread them again 😂

Reply:
done